### DISCUSSION TITLE
MODE output naming conventions

### DISCUSSION THREAD

**Discussion by mpm-meto** [dtcenter/METplus#2097] [File I/O] [answered]
One thing I find somewhat cumbersome is the output directory structure for running MvMODE. There is nothing wrong per se with the file/directory structure as laid out. It's pretty clear but out of context it's a bit more difficult to figure it out and can cause problems, as will be illustrated. 

Some of the output file names between the different cases (00,01,02) end up being the same which is fine when they are in their respective directories.  The super object output is suitably annotated with the variable name but the rest are not. 

Flicking between different dates (say), which is the top level output directory, requires navigating up and down the directory trees... I've tried to circumvent this by copying multiple days' output to another directory where all the files are in the same place but then I have another problem. Some of the files are overwritten or not copied. Is there a way of making the MODE output filename to always be more descriptive, i.e. add the variable into the filename as you do for the super object?  

e.g. *precipitation_amount_super_1200000L_20200706_000000V_240000A.ps is great....
Sub directories are:
00/mode_1200000L_20200706_000000V_000000A.ps 
01/mode_1200000L_20200706_000000V_000000A.ps 
02/mode_1200000L_20200706_000000V_240000A.ps
This is the standard MODE output convention. The output in 00 and 01 is called the same but one is wind (say) and the other is something else... the 02 directory contains an accumulation and is "safe" from being overwritten.

**Comment by georgemccabe** [comment, marked as answer]
Hi @mpm-meto, could you set the output prefix (MODE_OUTPUT_PREFIX) to include the variable name? This would add this info to the filename after the `mode_` prefix.

**Reply by JohnHalleyGotway** [reply]
Marion, just following up on this discussion. I definitely agree with George. Setting MODE_OUTPUT_PREFIX is the best/only way to customize the MODE output file naming convention. Does that suffice?

If so, please feel to click the "Mark as answer" button to close out the discussion. Thanks!
